utils/logger.py

```diff
@@ -66,7 +73,27 @@ def dict2path(my_dict, path=None):
 
 
 def dict_path(param_dict, param_name):
+    """
+    TODO
+    @param param_dict:
+    @param param_name:
+    @return:
+    """
     d2p = OmegaConf.to_container(param_dict[param_name], resolve=True)
     return {
         param_name + '.' + '.'.join(path): v for path, v in dict2path(d2p)
     }
+
+
+def get_logger(name=__name__, level=logging.INFO) -> logging.Logger:
+    """Initializes multi-GPU-friendly python logger."""
+
+    logger = logging.getLogger(name)
+    logger.setLevel(level)
+
+    # this ensures all logging levels get marked with the rank zero decorator
```

Comment:
cool!

Reply:
this was @CharlesAuthier addition though :)